unittests/savanna_misc_tests.cpp

```diff
@@ -791,4 +787,183 @@ BOOST_FIXTURE_TEST_CASE(finality_advancing_past_block_claimed_on_alternate_branc
+
+                                                             ^
+                                                             |
+                                             Validating the strong QC on b2 should
+                                             not fail for nodes which receive b4 and
+                                             b5 prior to b7 despite b5 advancing the
+                                             fork DB root to b3.
+
+Meaning of the superscripts and marks on the votes:
+The vote on block b was delayed in reaching the node for the producer p scheduled
+for the block at the next time slot t after block b by enough that a block produced on time by
+producer p for time slot t could not possibly utilize the vote in any QC the block could claim.
+Furthermore, the delay is such that the earliest time slot at which producer p could
+produce a block that utilizes the delayed vote is the time slot (t + d) where ...
+¹ ... d = 1.
+‡ ... d is infinite meaning the vote may never be received by producer p.
+
+steps mentioned in comments below refer to issue https://github.com/AntelopeIO/spring/issues/751
+
+Diagram below shows the timeline for nodes A, B, C and D receiving blocks b1 through b9.
+(x) marks the producer of the block.
+
+step    network partition        A        B        C        D
+---------------------------------------------------------------
+                                 b1       b1       b1(x)    b1
+(3)     A / B C D
+(4)                                       b2       b2(x)    b2
+(9)                                       b3       b3(x)    b3
+(15)    A / B C / D
+(18)                                      b4       b4(x)
+(20)                                                        b6(x)
+(22)    A D / B C
+(23)                             b2
+(25)                             b3
+(26)    A B C / D
+(28)                             b4
+(30)    A B / C / D
+(31)                                               b5(x)
+(33)                                                        b7(x)
+(35)    A B D / C
+(36)                             b6       b6
+(38)    A B C D
+(39)                             b5       b5
+(40)                                               b6
+(41,43)                          b7       b7       b7
+(44)                             b8       b8       b8       b8(x)
+(51)                             b9       b9       b9       b9(x)
+
+--------------------------------------------------------------------------------------------------------*/
+BOOST_FIXTURE_TEST_CASE(finality_advancing_past_block_claimed_on_alternate_branch, savanna_cluster::cluster_t) try {
+   using namespace savanna_cluster;
+   auto& A=_nodes[0]; auto& B=_nodes[1]; auto& C=_nodes[2]; auto& D=_nodes[3];
+
+   //_debug_mode = true;
+
+   auto b0 = A.produce_block();
+   print("b0", b0);
+
+   signed_block_ptr b1, b2, b3, b4, b5, b6, b7, b8, b9;
+
+   set_partition({ &A });
```

Comment:
Not every step, but maybe add comment indicating which code block corresponds to which range of steps specified in the issue?